lib/autoload/duplicator.rb

```diff
@@ -0,0 +1,26 @@
+class Duplicator
+  # Create an instance of Duplicator to track duplicated objects.
+  #
+  # @param [Array] excluded_objects A list of objects to be excluded from duplication
```

Comment:
...why not just accept an array or set or enumerable, then convert it to a set in the initialiser?